### PR TITLE
fix: Fix GetSignBytes with correct codec

### PR DESCRIPTION
**Pull request by amaurym**
## Description

While debugging regen-js, I noticed a bug in regen-ledger amino signing. Basically, in GetSignBytes, we're always using `ecocredit.ModuleCdc.MustMarshalJSON`. Unfortunately, this ModuleCdc doesn't get registered with any msgs, see [here](https://github.com/regen-network/regen-ledger/blob/00829dac260fa304760e399561f89eed15350a11/x/ecocredit/codec.go#L9).

This results in a level of nesting that's being cut-off when serializing into JSON:
```json
// expected
{
  "type": "regen/MsgSend",
  "value": {
    "recipient": "regen1abc...def"
  }
}
//actual
{
  "recipient": "regen1abc...def"
}
```

This bug also happened in the past in the SDK, read https://medium.com/confio/authz-and-ledger-signing-an-executive-summary-8754a0dc2a88 for more info.

The proposed fix is to make sure that the `ModuleCdc` that is used in GetSignBytes went through the modele's `RegisterLegacyAminoCodec`.

This is a consensus-breaking change. The same change needs to be done to all Msgs (I only did a poc on MsgSend). It's not a security issue, but more an annoyance, as without this change, the CosmJS+Amino+RegenLedger integration unfortunately won't work, see https://github.com/regen-network/regen-js/pull/41#discussion_r965192104.

Closes: #1481 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/base/types/v1/codec.go [hunk 1]
```diff
@@ -40,11 +39,11 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 }
 
 var (
-	amino = codec.NewLegacyAmino()
+	amino     = codec.NewLegacyAmino()
+	ModuleCdc = codec.NewAminoCodec(amino)
 )
 
 func init() {
-	RegisterLegacyAminoCodec(legacy.Cdc)
```
Comment (amaurym) [inline x/ecocredit/base/types/v1/codec.go:47]: We don't need to register the global legacy.Cdc, though it also doesn't hurt

**Review by technicallyty — APPROVED** [review]
lgtm, opened https://github.com/regen-network/regen-ledger/issues/1481 to follow up on the rest of the messages

**Comment by technicallyty** [comment]
@clevinson i'm just going to push the rest of the message fixes to this PR

**Review by clevinson — APPROVED** [review]
thanks for finding this @amaurym!!

File: x/ecocredit/base/types/v1/msg_send_test.go [hunk 2]
```diff
@@ -39,3 +39,12 @@ func (s *msgSend) ExpectTheError(a string) {
 func (s *msgSend) ExpectNoError() {
 	require.NoError(s.t, s.err)
 }
+
+func TestMsgSendAmino(t *testing.T) {
+	msg := &MsgSend{}
+	require.Equal(
+		t,
+		`{"type":"regen/MsgSend","value":{}}`, // And not only `{}` as on master
```
Comment (amaurym) [inline x/ecocredit/base/types/v1/msg_send_test.go:47]:
```suggestion
		`{"type":"regen/MsgSend","value":{}}`, // Make sure we have the `type` and `value` fields
```

**Review by aleem1314 — APPROVED** [review]
lgtm

File: CHANGELOG.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#1337](https://github.com/regen-network/regen-ledger/pull/1337) Add `AddCreditType` msg-based gov proposal
 - [#1346](https://github.com/regen-network/regen-ledger/pull/1346) Add `RemoveAllowedDenom` msg-based gov proposal
 - [#1349](https://github.com/regen-network/regen-ledger/pull/1349) Add `UpdateBasketFees` msg-based gov proposal
-  - [#1465](https://github.com/regen-network/regen-ledger/pull/1465) Renamed to `UpdateBasketFee`
+- [#1465](https://github.com/regen-network/regen-ledger/pull/1465) Renamed to `UpdateBasketFee`
```
Comment (ryanchristo) [inline CHANGELOG.md:79]: @technicallyty do you have an auto-format setting enabled? The spaces here and below were intentional.

**Comment by ryanchristo** [comment]
Thanks @amaurym and @technicallyty  :pray:

File: CHANGELOG.md (hunk 3 above)
Comment (technicallyty) [reply CHANGELOG.md:79]: oops, i thought those were formatting errors 🥲

Comment (ryanchristo) [reply CHANGELOG.md:79]: No problem. We can revise in another pull request.

**Pull request merged**